### PR TITLE
additionalData map to store material data

### PR DESCRIPTION
**Pull request by srinivasupadhya**

### CODE REVIEW

**Comment by srinivasupadhya** [comment]
Moved parsing response out so its done only once.

File: server/src/com/thoughtworks/go/server/service/materials/PluggableSCMMaterialPoller.java [hunk 1]
```diff
@@ -74,6 +96,19 @@ private void populateConfiguration(Configuration configuration, com.thoughtworks
         }
     }
 
+    private void updateAdditionalData(final long materialId, final Map<String, String> materialData) {
```
Comment (arvindsv) [inline server/src/com/thoughtworks/go/server/service/materials/PluggableSCMMaterialPoller.java]: Why wouldn't you do this inside MaterialRepository? I think it should.

Comment (srinivasupadhya) [inline server/src/com/thoughtworks/go/server/service/materials/PluggableSCMMaterialPoller.java]: same as [this](https://github.com/gocd/gocd/blob/4b838fed09498c01b0feb876080de7b74f9f18a8/server/src/com/thoughtworks/go/server/materials/PluggableSCMMaterialUpdater.java#L51) & [this](https://github.com/gocd/gocd/blob/7958ac8b613031660ee00f0591af792c06c0f74c/server/src/com/thoughtworks/go/server/materials/PackageMaterialUpdater.java#L51). I dont have any opinion, so if you think it should, will do :+1:

Comment (arvindsv) [inline server/src/com/thoughtworks/go/server/service/materials/PluggableSCMMaterialPoller.java]: :( I think everything that reads or write from material-related tables, should be somewhere in the MaterialRepository or around it.

It's your call, finally.

**Comment by ketan** [comment]
@srinivasupadhya 

Verified - works as expected.

**Pull request merged**